src/redux/books/books.js

```diff
@@ -1,35 +1,68 @@
-export const ADD_BOOK = 'book-store/books/ADD_BOOK';
-export const REMOVE_BOOK = 'book-store/books/REMOVE_BOOK';
+export const ADD_BOOK = 'bookstore-react/books/ADD_BOOK';
+export const GET_BOOKS = 'bookstore-react/books/GET_BOOKS';
+export const REMOVE_BOOK = 'bookstore-react/books/REMOVE_BOOK';
+export const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi';
+export const key = 'YT5Zn3abbZAmKMoCagyt';
 
-const initialState = [
-  { id: 0, Title: 'The Witcher', Author: 'Andrezj Sarpowsky' },
-  { id: 1, Title: 'Lord of The Rings', Author: 'Tolkein' },
-  { id: 2, Title: 'Percy Jackson', Author: 'Rick Rodini' },
-  { id: 3, Title: 'Harry Potter', Author: 'J.K Rowling' },
-  { id: 4, Title: 'Darwins Doubt', Author: 'John Meyers' }];
-
-const booksReducer = (state = initialState, action) => {
+const bookReducer = (state = [], action = {}) => {
   switch (action.type) {
     case ADD_BOOK:
+      return [...state, action.book];
+    case GET_BOOKS:
       return [...state, action.payload];
-
     case REMOVE_BOOK:
-      return state.filter((book) => book.id !== action.payload);
-
+      return state.filter((book) => book.item_id !== action.book.item_id);
     default:
       return state;
   }
 };
 
-const addBook = (payload) => ({
-  type: ADD_BOOK,
-  payload,
-});
+export const getBooks = () => async (dispatch) => {
+  const collectBooks = await fetch(`${url}/apps/${key}/books`).then((res) => res.json());
+  const books = [];
+  Object.keys(collectBooks).forEach((e) => books.push({ ...collectBooks[e][0], item_id: e }));
+  dispatch({
+    type: GET_BOOKS,
+    payload: books,
+  });
+};
+
+export const addBook = (book) => async (dispatch) => {
+  const data = {
+    item_id: book.item_id,
+    Title: book.Title,
+    Author: book.Author,
+    Category: book.Category,
+  };
```

Comment:
- [ ] Good job with this approach however the project requires that you use `createAsyncThunk`  for creating thunk action creators. Read a bit about [`createAsyncThunk`](https://redux-toolkit.js.org/api/createAsyncThunk) and how to achieve this. Also, please find the below code as an example of how you can use it in this case.
```

import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import axios from 'axios'; 


export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },
); 
```

- [ ] I recommend that you do the above for all the action creators.